account_payment_sale/tests/test_sale_order.py

```diff
@@ -8,16 +8,21 @@
 
 class TestSaleOrder(CommonTestCase):
     def create_sale_order(self, payment_mode=None):
-        with Form(self.env["sale.order"]) as sale_form:
-            sale_form.partner_id = self.base_partner
-            sale_form.pricelist_id = self.env.ref("product.list0")
-            for (_, p) in self.products.items():
-                with sale_form.order_line.new() as order_line:
-                    order_line.product_id = p
-                    order_line.name = p.name
-                    order_line.product_uom_qty = 2
-                    order_line.product_uom = p.uom_id
-                    order_line.price_unit = p.list_price
+        # Apply default to avoid error if user does not have
+        # product.group_product_pricelist group
+        sale_form = Form(
+            self.env["sale.order"].with_context(
+                default_pricelist_id=self.env.ref("product.list0").id
+            )
+        )
+        sale_form.partner_id = self.base_partner
```

Comment:
Here the pricelist is changed to the one of the partner, so the approach shouldn't be valid.

Reply:
Ok I understand, better then remove the `pricelist_id` assignment and leave the corresponding one when defining the partner.